docs/DockerContainerLifecycle.md

```diff
@@ -0,0 +1,340 @@
+
+    calicoctl container add <container_id> <ip>
+
+where `<container_id>` is the ID or name of the Docker container and `<ip>` 
+is an IP address falling within a configured Calico pool (eg. `192.168.1.1`), a 
+CIDR representing an existing Calico pool (eg. `192.168.0.0/16`), or the ip 
+address version to use (`ipv4` or `ipv6`).
+
+With this in mind, we run the following commands:
+
+    # Host1
+    calicoctl container add workload_A 192.168.1.1
+    
+    # Host2
+    calicoctl container add workload_B 192.168.1.2
+
+See the [`calicoctl container` reference](calicoctl/container.md) for a list of 
+all `calicoctl container` commands with their usages and examples.
+
+### Calico Actions
+When a workload is added to Calico using `calicoctl`, the tool checks the etcd 
+datastore to confirm that the IP value passed in coincides with a previously 
+defined IP pool. It then creates a veth pair on the host for the new endpoint, 
+storing one end in the host's network namespace and moving the other end into 
+the container.  This creates a new interface within the container.
+
+After successfully creating the veth pair, `calicoctl` saves all of the 
+follwoing data about the container within etcd:
+  - workload id 
+  - endpoint id
+  - the endpoint's assigned IP
+  - the host's IP
+  - the MAC address of the container's new veth interface
+  - the name of the veth on the host's end
+
+The are stored in etcd as:
+
+    # KEY
+    /calico/v1/host/Host1/workload/docker/<workload_id>/endpoint/<endpoint_id>
+    
+    # VALUE
+    {
+     "ipv6_gateway": null,
+     "state": "active",
+     "name": "<host_veth_name>",
+     "ipv4_gateway": "<host_ip>",
+     "ipv6_nets": [], "profile_ids": [],
+     "mac": "<container_mac>",
+     "ipv4_nets": ["<assigned_ip>"]
+    }
+
+Felix reads the changes in etcd, then programs routes to the container's 
+IP address via the host's veth.  Felix also, by default, programs ACLs into the 
+host's Linux kernel ip tables that drop all traffic to the container. A policy 
+profile will be used to configure inbound and outbound traffic rules later (see below).
+
+BIRD sees that new routes have been added to the host's routing table.  BIRD 
+distributes these routes to all of its BGP peers, such as Host2.
+
+### Result
+At this point there is still no connectivity between containers as no network 
+policy has been configured.
+
+
+![calicoctl container add](diagrams/container_add.png)
+
+## Create a Profile
+
+A profile represents a security policy that can be applied to any number of 
+containers in the Calico network.  A profile is required in order for nodes to 
+be able to communicate with one another and to access the internet.
+
+Profiles can be modified to allow or deny specific kinds of traffic, such as 
+`allow tcp from ports 80,443 tag PROFILE`.  Profiles have an implicit `deny` 
+rule at the end of the list of allow rules, so an `allow` rule is required for 
+all desired connections.
```

Comment:
Two suggestions:

1. "the list of allow rules" < isn't this just the list of rules?
2. I would be more explicit about what this means for the user:

  > Profiles have an implicit deny rule at the end of their rule list, so all desired connections must be explicitly permitted with an allow rule.